src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -231,6 +232,22 @@ doDiscard dbenv = runBlockEnv dbenv $ do
     --
     commitSavepoint Block
 
+doGetEarliest :: Db logger -> IO (Maybe (BlockHeight, BlockHash))
```

Comment:
This is just a clone of `doGetLatest`, but `ORDER BY ASC LIMIT 1` instead of `ORDER BY DESC LIMIT 1`.